src/main/java/com/notably/model/BlockModelImpl.java

```diff
@@ -0,0 +1,71 @@
+package com.notably.model;
+
+import com.notably.commons.core.path.AbsolutePath;
+import com.notably.model.block.Block;
+import com.notably.model.block.BlockImpl;
+import com.notably.model.block.BlockTree;
+import com.notably.model.block.BlockTreeImpl;
+import com.notably.model.block.Body;
+import com.notably.model.block.exceptions.CannotModifyRootException;
+import com.notably.model.block.exceptions.NoSuchBlockException;
+
+import javafx.beans.property.Property;
+
+/**
+ * The implementation class of BlockModel.
+ */
+public class BlockModelImpl implements BlockModel {
+    private BlockTree blockTree;
+    private Property<AbsolutePath> currentlyOpenPath;
+
+    public BlockModelImpl() {
+        blockTree = new BlockTreeImpl();
+    }
+
+    @Override
+    public BlockTree getBlockTree() {
+        return blockTree;
+    }
+
+    @Override
+    public AbsolutePath getCurrentlyOpenPath() {
+        return currentlyOpenPath.getValue();
+    }
+
+    @Override
+    public Property<AbsolutePath> getCurrentlyOpenPathProperty() {
+        return currentlyOpenPath;
+    }
+
+    @Override
+    public boolean hasPath(AbsolutePath p) {
+        try {
+            blockTree.get(p);
+            return true;
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+    @Override
+    public void setCurrentlyOpenBlock(AbsolutePath p) throws NoSuchBlockException {
+        blockTree.get(p);
```

Comment:
Why would you call this? To know whether or not the path exists? If that's the case, wouldn't it be better if you do the checking via `hasPath` instead?

Reply:
Got it! Thanks!
My initial intention was to take advantage of the `blockTree.get(p)` throwing the `NoBlockException` instead of having to throw a new one